app/controller/device_plant_controller.py

```diff
@@ -82,3 +87,31 @@ def get_device_plant_relation(req: Request, id_plant: str):
 @withSQLExceptionsHandle
 def get_all_device_plant_relations(req: Request, limit: int):
     return req.app.database.find_all(limit)
+
+
+@withSQLExceptionsHandle
+def delete_device_plant_relation(
+    req: Request,
+    response: Response,
+    id_device: Optional[str] = None,
+    id_plant: Optional[str] = None,
+):
+    if (id_device is None and id_plant is None) or (id_device and id_plant):
+        raise HTTPException(
+            status_code=status.HTTP_400_BAD_REQUEST,
+            detail="No identifier was provided, or both identifiers were provided.",
+        )
+
+    result_rowcount = 0
+    if id_device is not None:
+        result_rowcount = req.app.database.delete_by_field(
+            "id_device", id_device)
+    else:
+        result_rowcount = req.app.database.delete_by_field(
+            "id_plant", id_plant)
+
+    if result_rowcount == 0:
+        response.status_code = status.HTTP_204_NO_CONTENT
+        return  # EMPTY RESPONSE! RESOURCE DID NOT EXIST
+    else:
+        return "Device-plant relation deleted successfully"

```

Comment:
Wat, por que retornar un string que pinta mas como un log?

Reply:
Por default los strings se retornan directo en el body como json, y este endpoint por default retorna 200 en caso de exito. Pero ahi los retorno como [JSONResponse](https://fastapi.tiangolo.com/advanced/response-directly/#using-the-jsonable_encoder-in-a-response) para que quede más claro.

Reply:
Cambiado